web/src/stream_edit.js

```diff
@@ -92,6 +92,8 @@ function get_sub_for_target(target) {
     return sub;
 }
 
+let default_stream_previous_value = false;
```

Comment:
In the stream editing UI, we disable the `default stream` checkbox when the user switches to a `private` stream. Therefore, when the user switches back to a `public` stream without saving the changes, should the checkbox retain its previous value, similar to the `announce stream` or `is_default_stream` value of the stream?

Currently, the code supports the former behavior.

Reply:
I am not sure whether we just do not want the user to allow changing it to "private" stream if the stream is default unless the user does not mark the stream to not be default. @alya thoughts?

Reply:
Yeah, I guess we should prevent the user from creating an incompatible configuration. So for both stream creation and stream editing, we can:

- disable the private stream options if the default stream option is checked
- disable the default stream option if a private stream option is selected

And in both types of cases, we can show a tooltip:

> Private streams cannot be default streams.

I think that will be a nicer experience than showing an error after you try to make the changes, which is what we do now:

![Screen Shot 2023-05-23 at 5 32 04 PM](https://github.com/zulip/zulip/assets/2090066/d5a6ff74-79b4-4c9d-a49e-183c4e37aed0)